apps/aecore/lib/aecore/chain/block_validation.ex

```diff
@@ -100,34 +99,31 @@ defmodule Aecore.Chain.BlockValidation do
     valid_txs_list
   end
 
-  @spec validate_transaction_chainstate(%SignedTx{}, map()) :: {boolean(), map()}
+  @spec validate_transaction_chainstate(SignedTx.t, map) :: {boolean(), map}
   defp validate_transaction_chainstate(tx, chain_state) do
     chain_state_has_account = Map.has_key?(chain_state, tx.data.from_acc)
-    tx_has_valid_nonce = cond do
-      chain_state_has_account ->
-        tx.data.nonce > Map.get(chain_state, tx.data.from_acc).nonce
-
-      true ->
-        true
+    tx_has_valid_nonce = if chain_state_has_account do
+      tx.data.nonce > Map.get(chain_state, tx.data.from_acc).nonce
+    else
+      true
     end
 
     from_account_has_necessary_balance =
       chain_state_has_account &&
         chain_state[tx.data.from_acc].balance - (tx.data.value + tx.data.fee) >= 0
 
-    cond do
-      tx_has_valid_nonce && from_account_has_necessary_balance ->
-        from_acc_new_state = %{balance: -(tx.data.value + tx.data.fee), nonce: 1, locked: []}
-        to_acc_new_state = %{balance: tx.data.value, nonce: 0, locked: []}
-        chain_state_changes = %{tx.data.from_acc => from_acc_new_state, tx.data.to_acc => to_acc_new_state}
-        updated_chain_state = ChainState.calculate_chain_state(chain_state_changes, chain_state)
-        {true, updated_chain_state}
-      true ->
-        {false, chain_state}
+    if tx_has_valid_nonce && from_account_has_necessary_balance do
+      from_acc_new_state = %{balance: -(tx.data.value + tx.data.fee), nonce: 1, locked: []}
+      to_acc_new_state = %{balance: tx.data.value, nonce: 0, locked: []}
+      chain_state_changes = %{tx.data.from_acc => from_acc_new_state, tx.data.to_acc => to_acc_new_state}
+      updated_chain_state = ChainState.calculate_chain_state(chain_state_changes, chain_state)
+      {true, updated_chain_state}
+    else
+      {false, chain_state}
     end
   end
 
-  @spec calculate_root_hash(list()) :: binary()
```

Comment:
**Note:** erlang requires parens for its types, but elixir has them optional. I opted for the elixir style typenames without parens, but this should be a team choice. I can easily revert this change if desired.

:question: Which style does the team prefer; erlang style with parens (eg `binary()`) or elixir style without parens (eg `binary`)?

Reply:
We've used the erlang style so far so let's keep it like that